### PR TITLE
Add docs for Distributions.jl

### LINKED ISSUE #615: Document how to use Distributions.jl to create sample space
Hello @odow,

I am wondering if you would consider a feature to make it easier to parameterize the noise terms using types from Distributions.jl, barring any fundamental incompatibilities with the structure of SDDP.jl. I'm aware that the restriction is that the distributions be finite and discrete, but the API of Distributions.jl could check that easily (e.g. with `minimum` and `maximum` or by checking against "approved" types), and the type hierarchy can ensure that users only plug in discrete distributions. It would in particular make sampling multi-variate noise easier, with something like 

```julia
julia> Product([Bernoulli(0.5), Binomial(10,0.5), Binomial(50,0.25)])
Product{Discrete, Distribution{Univariate, Discrete}, Vector{Distribution{Univariate, Discrete}}}(v=Distribution{Univariate, Discrete}[Bernoulli{Float64}(p=0.5), Binomial{Float64}(n=10, p=0.5), Binomial{Float64}(n=50, p=0.25)])
```

If this is something possible and in scope for SDDP, I'd be happy to work on a PR to implement such functionality. A cursory glance at the source code seemed to indicate most of the relevant methods and types are in user_interface.jl, are there other places I should look too?

### PR DESCRIPTION
**Pull request by slwu89**
To address #615, update guides/add_multimensional_noise.md. I felt that the subject was essentially the same as this existing help file, and would rather add to it than create another one with a similar name. I listed two methods, sampling `N` points from the multidimensional space, and exhaustive enumeration. I used `IterTools.product` to generate the product space, I think this should be fairly minimal as a dependency as it has no dependencies of its own, and AFAIK is close to being a "base package" for Julia.

Closes #615

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -1,11 +1,13 @@
 [deps]
 CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
 DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+Distributions = "31c24e10-a181-5473-b8eb-7969acd0382f"
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
 ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
 GR = "28b8d3ca-fb5f-59d9-8090-bfdbd6d07a71"
 HiGHS = "87dc4568-4c63-4d18-b0c0-bb2238e4078b"
 Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
+IterTools = "c8e1da08-722c-5040-9ed9-7db0dc04731e"
```
Comment (odow) [inline docs/Project.toml:10]:
```suggestion
```

**Comment by odow** [comment]
> I used IterTools.product

`Base.product` is even simpler :smile:

**Comment by odow** [comment]
I made a few simple changes. Mainly just formatting. I think this should be pretty helpful. Now that I think about it, I've answered a similar question too many times. I always assumed that it was pretty obvious that you could use distributions and sample. But this makes things a lot clearer.

Any suggestions for other changes?

**Comment by slwu89** [comment]
Did not know about `Base.product`, that's fantastic. No other suggestions, thanks!

File: docs/src/guides/add_multidimensional_noise.md [hunk 2]
```diff
@@ -28,3 +35,94 @@ julia> SDDP.simulate(model, 1);
...
+
+There are several ways to go about this. If the sample space is finite, and
+small enough that it makes sense to enumerate each element, we can use
+`Base.product` and `Distributions.support` to generate the entire sample space
+`Ω` from each of the marginal distributions. 
+
+We can then evaluate the density function of the product distribution on each
+element of this space to get the vector of corresponding probabilities, `P`.
+
+```jldoctest; filter=[r"\(\d, \d, \d\)", r"1\-element.+"s]
```
Comment (odow) [inline docs/src/guides/add_multidimensional_noise.md:54]:
```suggestion
```jldoctest; filter=[r"\[\d+, \d+, \d+\]", r"1\-element.+"s]
```

File: docs/src/guides/add_multidimensional_noise.md [hunk 3]
```diff
@@ -28,3 +35,94 @@ julia> SDDP.simulate(model, 1);
...
+
+### Sampling
+
+For sample spaces that are too large to explicitly represent, we can instead
+approximate the distribution by a sample of `N` points. Now `Ω` is a sample from
+the full sample space, and `P` is the uniform distribution over those points.
+Points with higher density in the full sample space will appear more frequently
+in `Ω`.
+
+```jldoctest; filter=[r"\[\d, \d, \d\]", r"1\-element.+"s]
```
Comment (odow) [inline docs/src/guides/add_multidimensional_noise.md:96]:
```suggestion
```jldoctest; filter=[r"\[\d+, \d+, \d+\]", r"1\-element.+"s]
```

File: docs/src/guides/add_multidimensional_noise.md [hunk 4]
```diff
@@ -9,14 +9,21 @@ end
 Multi-dimensional stagewise-independent random variables can be created by
 forming the Cartesian product of the random variables.
 
+## A simple examle
```
Comment (odow) [inline docs/src/guides/add_multidimensional_noise.md:12]:
```suggestion
## A simple example
```

**Review by odow — APPROVED** [review]

**Pull request merged**